pkgs/development/tools/rufo/default.nix

```diff
@@ -0,0 +1,16 @@
+{ bundlerEnv, bundlerUpdateScript, ruby, lib }:
+
+bundlerEnv {
```

Comment:
Maybe `bundlerApp` is more approapriate to use here than `bundlerEnv`, as [noted in the manual](https://nixos.org/manual/nixpkgs/stable/#sec-language-ruby)?  I'm seeing this tree exposing both `bundler` and `rufo` (though only `rufo` can be called in `$PATH`,) built during `nixpkgs-review`:

```sh
[zakame:~/.cache … -review/pr-100663] % tree results/rufo/
results/rufo/
├── bin
│   └── rufo
└── lib
    └── ruby
        └── gems
            └── 2.6.0
                ├── bin
                │   ├── bundle -> /nix/store/3h2mgk5jbg33kxif9c4girl25v45h5cd-bundler-2.1.4/lib/ruby/gems/2.6.0/bin/bundle
                │   ├── bundler -> /nix/store/3h2mgk5jbg33kxif9c4girl25v45h5cd-bundler-2.1.4/lib/ruby/gems/2.6.0/bin/bundler
                │   └── rufo -> /nix/store/bcsv8k1q77piw33mvpjafjk3h186myhc-ruby2.6.6-rufo-0.12.0/lib/ruby/gems/2.6.0/bin/rufo
                ├── build_info
                ├── doc
                ├── extensions
                ├── gems
                │   ├── bundler-2.1.4 -> /nix/store/3h2mgk5jbg33kxif9c4girl25v45h5cd-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4
                │   └── rufo-0.12.0 -> /nix/store/bcsv8k1q77piw33mvpjafjk3h186myhc-ruby2.6.6-rufo-0.12.0/lib/ruby/gems/2.6.0/gems/rufo-0.12.0
                └── specifications
                    ├── bundler-2.1.4.gemspec -> /nix/store/3h2mgk5jbg33kxif9c4girl25v45h5cd-bundler-2.1.4/lib/ruby/gems/2.6.0/specifications/bundler-2.1.4.gemspec
                    └── rufo-0.12.0.gemspec -> /nix/store/bcsv8k1q77piw33mvpjafjk3h186myhc-ruby2.6.6-rufo-0.12.0/lib/ruby/gems/2.6.0/specifications/rufo-0.12.0.gemspec

13 directories, 6 files
```

Another option might be to just add a corresponding entry to https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/ruby-packages.nix to leverage the [ruby.withPackages](https://github.com/NixOS/nixpkgs/blob/547b5b7a73e8a780b6438cf1128388dbc51b002e/pkgs/development/ruby-modules/with-packages/default.nix) infrastructure (this is how [rubocop](https://github.com/NixOS/nixpkgs/blob/8dd059e344f7903e7aa0c9e19d60d2eb711e28e4/pkgs/top-level/ruby-packages.nix#L2157-L2167) is defined, for example.)

See also https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/ruby.section.md (although I wonder why this could not be found under the normal https://nixos.org/manuals/nixpkgs/unstable link...)

Reply:
Actually RuboCop is defined twice, in both ways: `pkgs.rubocop` uses `bundlerEnv`, while `pkgs.rubyPackages.rubocop` uses the `withPackages` system.

It looks like the `withPackages` system is intended to updated by a [script](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update-ruby-packages), not edited manually, which would mean updating every package in order to add one. It also seems it’s not really being used or [maintained](https://github.com/NixOS/nixpkgs/commits/master/pkgs/top-level/ruby-packages.nix).

So I’ve updated this using `bundlerApp`. Thanks!

Reply:
Yeah, I think `ruby.withPackages` is relatively new and hasn't taken off much yet, based on the [PR discussion](https://github.com/NixOS/nixpkgs/pull/61114).